anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/cusum/CusumAnomalyDetector.java

```diff
@@ -39,14 +39,11 @@
  * @author kashah
  */
 @Data
-public final class CusumAnomalyDetector implements AnomalyDetector {
+public final class CusumAnomalyDetector extends BasicAnomalyDetector<CusumParams> {
```

Comment:
I wouldn't expect model internals (i.e. `CusumParams`) to make it all the way up to the detector generic params, just based on [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) considerations. Detector can know about models but model params are too deeper than what the detector should know or care about. I think what we want here is for the model to appear as a generic param, and push all the param stuff down into the actual model.